forum/models.py

```diff
@@ -0,0 +1,68 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+from common.utils import create_code, create_slug
+
+class Forum(models.Model):
+	class Meta:
+		db_table = 'forums'
+
+	date_created = models.DateTimeField(auto_now_add=True)
+	date_modified = models.DateTimeField(auto_now=True)
+
+	code = models.CharField(max_length=60, unique=True)
+	slug = models.SlugField(max_length=100, unique=True)
+
+	created_by = models.ForeignKey(User, on_delete=models.CASCADE)
+	subject = models.CharField(max_length=200)
+	description = models.TextField(max_length=500, blank=True)
+
+	messages = models.ManyToManyField('Message', blank=True, through='ForumMessage')
+
+	def __str__(self):
+		return self.subject
+
+	def save(self, *args, **kwargs):
+		if self.pk is None:
+			self.code = create_code(self, 50)	
+			self.slug = create_slug(self, field_name='subject')
+		super(Forum, self).save(*args, **kwargs)
+
```

Comment:
need to be tested

Reply:
@Chamane  you have to test it . This is what you have to do in your reviews

Reply:
@Chamane  : Please make a **functional testing or exploratory testing first** . The unit test will come after your unit test example .